org-super-agenda.el

```diff
@@ -1099,6 +1114,53 @@ actually the ORDER for the groups."
        (-remove #'s-blank-str? it)
        (s-join "\n" it)))
 
+;;;; Hide/Show filter
+(defun org-super-agenda--hide-or-show-groups (&rest _)
+  "Hide/Show any empty/non-empty groups after `org-agenda-finalize',
+`org-agenda-filter-apply'  was called."
+  (cl-labels ((header-p () (org-get-at-bol 'org-super-agenda-header))
+              (grid-p () (not (or  (org-get-at-bol 'org-agenda-structural-header)
```

Comment:
It might be better to get the list of text properties at that position and compare it against a list of properties, rather than calling `org-get-at-bol` 4 times.  I'm not sure which would be faster.  Would you be willing to benchmark that?

Reply:
Hmm, not sure if this is the correct way to benchmark it!?

``` emacs-lisp
(benchmark 10000000 (not (let ((props (text-properties-at (point-at-bol))))
                           (or (member 'org-agenda-structural-header props)
                               (member 'org-agenda-date-header props)
                               (member 'type props)
                               (member 'org-super-agenda-header props)))))
"Elapsed time: 0.070366s"

"Elapsed time: 0.069554s"

"Elapsed time: 0.088570s"

"Elapsed time: 0.063170s"


(benchmark 10000000 (not (or (org-get-at-bol 'org-agenda-structural-header)
                             (org-get-at-bol 'org-agenda-date-header)
                             (org-get-at-bol 'type)
                             (org-get-at-bol 'org-super-agenda-header))))
"Elapsed time: 0.064548s"

"Elapsed time: 0.056398s"

"Elapsed time: 0.052549s"

"Elapsed time: 0.050331s"
```

Thanks!

Reply:
Not a large difference, but `org-get-at-bol` is faster.  Good to know.  Thanks.

Reply:
Actually, I just remembered that your benchmark doesn't use `lexical-binding`, which may make a difference.  Please try with the `bench-multi-lexical` macro from https://github.com/alphapapa/emacs-package-dev-handbook.

Reply:
``` emacs-lisp
(bench-multi-lexical :times 500000
  :ensure-equal t
  :forms (("test-properties-at" (not (let ((props (text-properties-at (point-at-bol))))
                                       (or (member 'org-agenda-structural-header props)
                                           (member 'org-agenda-date-header props)
                                           (member 'type props)
                                           (member 'org-super-agenda-header props)))))
          ("org-get-at-bol" (not (or (org-get-at-bol 'org-agenda-structural-header)
                                     (org-get-at-bol 'org-agenda-date-header)
                                     (org-get-at-bol 'type)
                                     (org-get-at-bol 'org-super-agenda-header))))))
```
Result:

 Form | x faster than next | Total runtime |  # of GCs | Total GC runtime 
 ----- | ---------------------- | ----------------- | ------------ | -----------------------
 test-properties-at | 2.39 | 0.150025 | 0 | 0 
 org-get-at-bol | slowest | 0.359302 | 0 | 0 


Seems like `org-get-at-bol` is now slower. Hmmm, interesting :-

Reply:
Ok, it uses `text-properties-at` now.